### PR TITLE
Reduce peak memory consumption during Datumaro and COCO extractors

### PR DESCRIPTION
**Pull request by wonjuleee**
### Summary
For importing COCO data, 
As-is: 
image 

To-be: 
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/unit/data_formats/datumaro/test_datumaro_format.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def _test_save_and_load(
         [
             pytest.param(
                 "fxt_test_datumaro_format_dataset",
-                compare_datasets_strict,
+                compare_datasets,
```
Comment (vinnamkim) [inline tests/unit/data_formats/datumaro/test_datumaro_format.py:67]: To replace `compare_datasets_strict` perfectly, we need `require_media=True` argument for `compare_datasets(a, b, require_media=True)`. Without this flag, `compare_datasets` do not check equivalence between the media entities.

Comment (wonjuleee) [reply tests/unit/data_formats/datumaro/test_datumaro_format.py:67]: Thank you for letting me know this. But below True is for require_media as a compare keyword argument.

Comment (vinnamkim) [reply tests/unit/data_formats/datumaro/test_datumaro_format.py:67]: I got it, but L206 still needs it.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by sooahleex — APPROVED** [review]
Looks good to me.

**Pull request merged**